### PR TITLE
Implement functionality to delete posts

### PR DESCRIPTION
**Pull request by Andrewmob2703-zz**

### CODE REVIEW

**Review by smarquez-jpg — APPROVED** [review]
LGTM

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 1]
```diff
@@ -35,12 +35,14 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    long messageId = Long.parseLong(request.getParameter("messageId"));
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: String constants like "messageId" would preferably be private/public static final class member variables.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 2]
```diff
@@ -35,12 +35,14 @@
...
+
+    Key postEntityKey = KeyFactory.createKey("blogMessage", messageId);
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: Maybe "blogMessage" key string should also be treated as a constant variable?

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 3]
```diff
@@ -35,12 +35,14 @@
...
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      datastore.delete(entity.getKey());
-    }
-    response.sendRedirect("/index.html");
+    datastore.delete(postEntityKey);
+
+    response.setContentType("text/html");
+    response.getWriter().println("Success! Post Deleted.");
+    return;
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: No need to do return at the end of a method.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -118,14 +118,19 @@ function createImgElement(text) {
...
+
+// Deletes a post and sends a confirmation message for 5 seconds.
+async function deleteMessage(msg) { 
   const params = new URLSearchParams();
   params.append('messageId', msg.id);
   fetch('/delete-data', {method: 'POST', body: params})
         .then(response => response.text()).then((text) => {
     const confirmationElement = document.getElementById('confirm');
-    confirmationElement.innerHTML = text;
+    confirmationElement.innerHTML="";
```
Comment (peizhaoou) [inline portfolio/src/main/webapp/script.js:129]: Put space before and after "="

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -118,14 +118,19 @@ function createImgElement(text) {
...
+    confirmationElement.style.display='inline-block';
+    confirmationElement.innerHTML = `<i>${text}</i>`;
+    setTimeout(function () {
+        document.getElementById('confirm').style.display='none';}, 5000);
```
Comment (peizhaoou) [inline portfolio/src/main/webapp/script.js:133]: 2-space indentation?

**Review by peizhaoou — APPROVED** [review]
LGTM overall. Thanks!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (Andrewmob2703-zz) [reply portfolio/src/main/webapp/script.js:133]: this function itself is a parameter so the space was meant to break the line. I dunno if that makes a difference...?

Comment (peizhaoou) [reply portfolio/src/main/webapp/script.js:133]: Okay thanks!

**Pull request merged**